src/modules/direction/components/PostTypesFilter.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useCallback } from 'react';
+import _ from 'lodash';
+import { useSelector, useDispatch } from 'react-redux';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import { useForm, Controller } from 'react-hook-form';
+import Checkbox from '@material-ui/core/Checkbox';
+import { RootStateType } from '../../../store/rootReducer';
+import { setPostFilters } from '../store/directionSlice';
+import { DirectionFilterTypes } from '../../../lib/types';
+
+export interface IPostTypeFilterProps {
+  directionName: string;
+}
+
+export const PostTypeFilter: React.FC<IPostTypeFilterProps> = ({
+  directionName,
+}) => {
+  const dispatch = useDispatch();
+  const { control } = useForm();
+
+  const postTypes = useSelector(
+    (state: RootStateType) => state.properties.postTypes,
+  );
+
+  const state = () => {
+    const initialTypeState = {};
+    postTypes?.forEach((type) => {
+      initialTypeState[type.id.toString()] = false;
+    });
+    return initialTypeState;
+  };
```

Comment:
this is `reduce` structure :)
```
const variable = {};
someArr.forEach(entry => variable[entry.id] = doSmth(entry));
return varuable;
```

```suggestion
  const state = () => postTypes?.reduce((acc, next) => acc[next.id.toString()], {}));
```

Reply:
done